sim-lib/src/random_activity.rs

```diff
@@ -94,12 +125,14 @@ impl Display for NetworkGraphView {
 /// node's capacity, it will produce payments such that the node sends multiplier * capacity over a calendar month.
 /// While the expected amount to be sent in a month and the mean payment amount are set, the generator will introduce
 /// randomness both in the time between events and the variance of payment amounts sent to mimic more realistic
-/// payment flows.
+/// payment flows. For deterministic time between events and payment amounts, the `RandomPaymentActivity` keeps a
+/// handle on a seed.
 pub struct RandomPaymentActivity {
     multiplier: f64,
     expected_payment_amt: u64,
     source_capacity: u64,
     event_dist: Exp<f64>,
+    seeded_rng: SeededRng,
```

Comment:
Not ideal in terms of abstraction, but I think we're going to need to create a _single_ RNG from the fixed seed, and provide it to every instance of `RandomPaymentActivity` that we have. Because as is, we create _multiple_ rngs with the _same_ seed, so each node will execute payments with the same series of rng values (if you have nodes with the same capacity, they all end up sending the same amount for their payments). 

[Tried this out here](https://github.com/user-attachments/files/16014458/single_rng.patch).

Reply:
I have made some changes so that all instances of `RandomPaymentActivity` share a single RNG but the deterministic outcome we want is not consistent. This could be down to how I have set my nodes but there is some `undesired` variance in the payment amount generated after multiple simulator runs.

Here is a small [table ](https://docs.google.com/spreadsheets/d/1MvN2tRgiOndapyyLi-gVFfhyhNVzC6TUKoETSsXavIk/edit?usp=sharing)showing the simulation run count and the payment amount generated at each run. The table is being updated every time I run the simulator to identify if the variance is as a result of my node setup or the code.

Please try this on your PC to see if you get similar outcomes.

Reply:
I think this is to be expected - tasks will complete in slightly different orders on each run (and payments will complete in different amounts of time), so they'll grab the common RNG is different orders. We can't make promises for totally deterministic runs, but I think that this is a good shot at it because it'll aggregate out over several runs. 

But well observed, let's make a note of it in the readme that we're not promising fully deterministic simulations.

Reply:
Can also confirm this by just writing a small unit test for `MutRng::new` and asserting that we get the same values for the same seeds.